### ISSUE TITLE
USER ISSUE: Building on ramps

### ISSUE THREAD

**Issue by chuntatchan** [StrangeLoopGames/EcoIssues#7347]
**Version:** 0.7.3.3 beta

**Steps to Reproduce:**
1. Go to First Person Mode.
2. Build a ramp on a flat surface.
3. Build a platform that's not touching the ramp 1 block high, on the same level as the flat surface.
4. Build a (can-be-built-upon) block two blocks high that's between the ramp and yourself.
5. Move really close to the buildable block to a point where you can see both the buildable block and the ramp intersecting each other.
6. Very slowly, attempt to aim at the fine line between the buildable block and the ramp and move the cursor back and forth till you can build a block on top of the ramp.

Perhaps the player was in a state that allowed for buildable material to be placed down on the surface that's being currenly's aimed at and bypassed a isBuildableSurface check or something.

Here's a video demonstrating the bug: https://drive.google.com/open?id=1CfOBqo0YVK9pQxYpcYneguk54tLM573p

**Expected behavior:**
It should have prevented the player from building on the ramp normally.

**Actual behavior:**
Instead it caused a bug allowing the player to build on ramps.

**Comment by fillion07** [comment]
I don't think this is a bug. I like the fact to be able to put block on top of a ramp. By example, I could want to put construction posts on the ramp to alert other vehicles that the road is not finished.

**Comment by metachronism** [comment]
passing to sean -
an interactionmanager/targeting thing, to me it still seems like its picking the right result though, so lowering priority
